### ISSUE TITLE
Fluent UI Coachmark component's teardrop is not accessible from keyboard

### ISSUE THREAD

**Issue by sergul** [microsoft/fluentui#15376]
### Environment Information

- **Package version(s)**: (7.123.1)
- **Browser and OS versions**: (Chromium Edge 85.0.564.68 and Windows 10)

### Describe the issue:

Coachmark component's teardrop element is not accessible by pressing tab key using Narrator screen reader.
In other words, when using keyboard by pressing tab, Narrator doesn't land / focus on Coachmark's teardrop element.

### Please provide a reproduction of the issue in a codepen:

1. Go to https://codepen.io/sergulo/pen/yLJBoEK using Chromium Edge browser
2. Turn on Narrator screen reader
2. Click on "Show coachmark" button and wait until the teardrop element is popped out
3. Press tab key on keyboard that Narrator could land / focus the teardrop element

#### Actual behavior:

When pressing tab key on keyboard Narrator doesn't land / focus on Coachmark's teardrop element

#### Expected behavior:

When pressing tab key on keyboard Narrator should land / focus on Coachmark's teardrop element

### Documentation describing expected behavior

https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc=%7b8492c4eb-c179-40ae-8777-cd044ed725a2%7d

**Comment by tomi-msft** [comment]
Hi @sergul Thanks for the issue! We will look into this more.

**Comment by paulgildea** [comment]
@smhigley FYI

**Comment by mimi-msft** [comment]
Hey @tomi-msft @paulgildea 

Do you have an update on when this will be fixed? We have a "Red" (most severe) level bug which we can't resolve until this is fixed!

Thanks,
Mimi

**Comment by tomi-msft** [comment]
Hey @mimi-msft thanks for your patience. I am having trouble reproing the bug. Opening the codepen with Narrator on, I am able to navigate to the coachmark, press enter, and the coachmark gets selected, and I am able to press enter and access it. Could you give some more details as to where you are having the issue?

**Comment by adam-msft** [comment]
I can repro the issue on the codepen and if you open support.xbox.com in an inprivate browser (to mimic being a first-time visitor). In both examples, if I use tab navigation, I cannot land on the teardrop/coachmark by tabbing through the page. I go through all other elements but not the coachmark.

**Comment by tomi-msft** [comment]
Hi @adam-msft, I'm sorry but I'm still unable to repro. For the codepen, I am able to tab to open coachmark, and when I press enter the focus gets automatically set to the coachmark, and then I am able to open it with enter or alt + c. For the support.xbox.com I am also able to open the coachmark with alt + c when I am tabbed over the target button (New in support).

**Comment by adam-msft** [comment]
Is the intended design for the coachmark that it can be accessible via tab navigation or that it can be accessible via alt+c? Tab navigation is what our accessibility team is asking for and what is missing on our support.xbox.com site right now.

**Comment by tomi-msft** [comment]
Hi @adam-msft, after further investigation, it seems like the problem is that the teardrop is wrapped within a Layer, which causes it to appear at the bottom of the DOM tree. The presence of the Layer wrapper is determined by the doNotLayer prop. In the codepen, if you set the doNotLayer prop to true, you should be able to tab to the Coachmark. This can be a temporary workaround for now, and I will get a more permanent fix out shortly!

**Comment by sergul** [comment]
Hi @tomi-msft,

Thanks for the quick workaround. Should this prop "doNotLayer" be set on "Coachmark" component ? If so then I don't see any difference in codeped with or without it.

https://codepen.io/sergulo/pen/yLJBoEK

**Comment by tomi-msft** [comment]
Hi @sergul, the doNotLayer prop is part of the positioningContainerProps, on L56 in your codepen. Try changing that to true. The other thing I will try to fix is that when the teardrop is tabbed to, there is no indication that it is in focus, so I will add styling to show that.

**Comment by sergul** [comment]
Great ! it's working now 😉 And yes we'd need to border it when it's tabbed. Thanks @tomi-msft

**Comment by sergul** [comment]
😞 @tomi-msft sorry for getting excited too soon but looks like it's still not working when Narrator screen reader is on. It was working without the screen reader but when I tabbed when Narrator is on, the focus is not landing on teardrop.

**Comment by tomi-msft** [comment]
@sergul I'm having a little trouble reproing. In the codepen when I tab over to the button to show the coachmark and hit enter, the teardrop automatically receives focus and the narrator picks that up. The issue I see is that if I don't open the teardrop after the initial callout, and I tab away, I can tab back to the teardrop, but the narrator no longer announces it. Is this what you're talking about?

**Comment by sergul** [comment]
@tomi-msft interesting but the Narrator's behavior is different on my side from what you described above.

When I tab over the button and hit enter, Narrator announces "a coachmark has appeared" but the teardrop doesn't automatically receives the focus.  Instead I see this. And when I try to tab again to highlight the teardrop it just ignores to land on it.
![image](https://user-images.githubusercontent.com/2910631/97628702-4bd19480-19ea-11eb-9dce-955d0679da89.png)

**Comment by tomi-msft** [comment]
@sergul Just to make sure I understand, you hit the `Show Coachmark` button, and the Narrator announces that a Coachmark has appeared, but when you press enter or alt + c you are unable to open it?

**Comment by sergul** [comment]
@tomi-msft yes that’s correct. Although I tried only with enter press but not with alt +c

Can test it a little later and update.

**Comment by sergul** [comment]
@tomi-msft so I just tested with Alt + c and instead of pressing tab key and the teardrop receives the focus as expected. By the way with using Alt + c I didn't have to even set "doNotLayer" to true.

@adam-msft do you think using only Alt + c to focus the coachmark teardrop covers our expected result ?

**Comment by sergul** [comment]
@tomi-msft we are curious if there is a way to change the keyboard combination of accessing the coachmark teardrop from Alt + c to e.g. CTRL+F6 ?

cc: @adam-msft, @mimi-msft

**Comment by jendowns** [comment]
Hello @tomi-msft! I noticed above in https://github.com/microsoft/fluentui/issues/15376#issuecomment-718976837 that you mentioned the lack of a focus indicator, and said you'd look into adding styles for it.

Are you working on that focus indicator here, as part of this issue? I'm asking because we have a focus indicator bug for the `CoachMark` filed in our ADO, and wanted to make sure we're tracking the right GitHub issue here. Thank you!

**Comment by mltejera** [comment]
+1 to @jendowns's comment.

**Comment by tomi-msft** [comment]
Hi @jendowns, I think it would be best to file a separate issue in github for that particular bug
